sphinx/environment/adapters/toctree.py

```diff
@@ -194,12 +194,11 @@ def _entries_from_toctree(toctreenode: addnodes.toctree, parents: List[str],
                     for subtocnode in list(toc.findall(addnodes.toctree)):
                         if not (subtocnode.get('hidden', False) and
                                 not includehidden):
-                            i = subtocnode.parent.index(subtocnode) + 1
-                            for entry in _entries_from_toctree(
+                            start = subtocnode.parent.index(subtocnode) + 1
+                            for i, entry in enumerate(_entries_from_toctree(
                                     subtocnode, [refdoc] + parents,
-                                    subtree=True):
+                                    subtree=True), start):
```

Comment:
I feel this is a bit unreadable. How about storing the list before this loop?
```
entries = _entries_from_toctree(...)
for i, entry in enumerate(entries, start):
    ...
```

Reply:
Sorry, it's my wrong. Okay, let's revert this change and go with your idea.

Reply:
i mean, it's a code smell. The fact that this *can't* be refactored in this way tells me that this *should* be refactored.
Not in this PR though.